python_modules/dagster/dagster_tests/core_tests/pending_repo_from_cached_asset_metadata.py

```diff
@@ -0,0 +1,76 @@
+from typing import cast
+
+from dagster import asset, define_asset_job
+from dagster._core.definitions.cacheable_assets import (
+    CACHED_ASSET_ID_KEY,
+    CACHED_ASSET_METADATA_KEY,
+    extract_from_current_repository_load_data,
+)
+from dagster._core.definitions.definitions_class import Definitions
+from dagster._core.definitions.repository_definition.repository_definition import (
+    PendingRepositoryDefinition,
+)
+from dagster._core.execution.plan.external_step import local_external_step_launcher
+from dagster._core.instance import DagsterInstance
+
+FETCHED_KVS_KEY = "fetched_external_data"
+USED_CACHE_KVS_KEY = "used_cached_external_data"
+
+instance = DagsterInstance.get()
+metadata_value_cached_assets = extract_from_current_repository_load_data("my_cached_asset_id")
+
+if metadata_value_cached_assets is not None:
+    get_definitions_called = int(
+        instance.run_storage.get_cursor_values({USED_CACHE_KVS_KEY}).get(USED_CACHE_KVS_KEY, "0")
+    )
+    instance.run_storage.set_cursor_values({USED_CACHE_KVS_KEY: str(get_definitions_called + 1)})
+else:
+    get_definitions_called = int(
+        instance.run_storage.get_cursor_values({FETCHED_KVS_KEY}).get(FETCHED_KVS_KEY, "0")
+    )
+    instance.run_storage.set_cursor_values({FETCHED_KVS_KEY: str(get_definitions_called + 1)})
+
+    metadata_value_cached_assets = [{"foo": "bar"}, {"baz": "qux"}]
+
+assert metadata_value_cached_assets == [
+    {"foo": "bar"},
+    {"baz": "qux"},
+], metadata_value_cached_assets
+
+
+@asset(
+    metadata={
+        CACHED_ASSET_ID_KEY: "my_cached_asset_id",
+        CACHED_ASSET_METADATA_KEY: {"foo": "bar"},
```

Comment:
so something that threw me for a bit more of a loop than it should have is the fact that this metadata is hardcoded on these assets, and so the stuff that's loaded from the repository load data is irrelevant.

I can see that the tests below are still valid, but I think this would make a bit more sense if you just wrote a function that returns assets based on the cached metadata, just to be a bit more "realistic"

Reply:
Updated